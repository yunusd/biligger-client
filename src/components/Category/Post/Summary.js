import React from 'react';
import { useApolloClient } from 'react-apollo-hooks';
import marked from 'marked';
import moment from 'moment';
import { Link } from 'react-router-dom';

import {
 Grid, Card, Icon, Segment,
} from 'semantic-ui-react';
import dateLocale from '../../../helpers/dateLocale';
import { GET_ME_FROM_CACHE, GET_AUTH_STATUS } from '../../../queries';

moment.updateLocale('en', dateLocale);

const Summary = (props) => {
  const client = useApolloClient();

  const { currentUser } = client.readQuery({ query: GET_AUTH_STATUS });
  const { getMe } = currentUser.isLoggedIn ? client.readQuery({ query: GET_ME_FROM_CACHE }) : false;


  const {
    // eslint-disable-next-line react/prop-types
    data, error, isExist, category,
  } = props;
  const posts = data.getLatestPostsByCategory;

  const colors = {
    teknoloji: category === 'teknoloji' && 'blue',
    bilim: category === 'bilim' && 'red',
    spor: category === 'spor' && 'green',
    sanat: category === 'sanat' && 'yellow',
    yasam: category === 'yaşam biçimi' && 'purple',
  };

  return (
    <React.Fragment>
      <Grid columns={2} centered>
        <Grid.Row>
          <Grid.Column width={12}>
            <Segment
              color={
                colors.teknoloji || colors.bilim
                || colors.spor || colors.sanat
                || colors.yasam
              }
              style={{
                boxShadow: 'none',
                border: 'none',
                fontWeight: '600',
                textTransform: 'capitalize',
              }}
            >
              { category }
            </Segment>
          </Grid.Column>
        </Grid.Row>
        {
          !isExist ? <h5>Bilig bulunamadı!</h5>
          : posts.map((val) => {

              const auth = {
                isOwn: getMe && getMe.username === val.author.username,
                isLoggedIn: currentUser.isLoggedIn && true,
              };

              const title = val.title.length < 100 ? val.title : val.title.slice(0, 100);
              const url = `${title.toLowerCase().replace(/\s/g, '-')}-${val.id}`;
              const rawContent = marked(val.content);

              const paragraph = rawContent.substring(
                rawContent.lastIndexOf('<p>'),
                rawContent.lastIndexOf('</p>'),
              );

              const content = paragraph.length < 500 ? paragraph : `${paragraph.slice(0, 500)}...`;

              return (
                <Grid.Row id={val.id} key={val.id}>
                  {error && 'HATA'}
                  <Grid.Column width={12}>
                    <Card fluid>
                      <Card.Content>
                        <Card.Header>
                          <Link
                            to={{
                              pathname: val.id ? `/${url}` : '/',
                              state: { id: val.id },
                            }}
                            style={{ color: 'rgba(0,0,0,.85)' }}
                          >
                            {title}
                          </Link>
                        </Card.Header>
                        <Card.Meta>
                          {val.author.username}
                            &nbsp;-&nbsp;
                          {moment(val.createdAt).fromNow()}
                        </Card.Meta>
                        <Card.Description dangerouslySetInnerHTML={{ __html: content }} className="display-linebreak">
                          {/* {content} */}
                        </Card.Description>
                      </Card.Content>

                      <Card.Content extra>
                        <Link to="/">
                          <Icon name="idea" />
                            Katılıyorum
                        </Link>

                        &nbsp;&nbsp;&nbsp;

                        <Link to="/">
                          <Icon name="comment" />
                            Yorum Yaz
                        </Link>

                        {auth.isLoggedIn && (
                            auth.isOwn ? (
                              <React.Fragment>
                                <Link to={`${url}/düzenle`} className="summary-context-right">
                                  <Icon name="edit" />
                                  Düzenle
                                </Link>
                              </React.Fragment>
                            ) : (
                              <Link to="#" className="summary-context-right" title="Bildir">
                                <Icon name="flag" title="bildir" />
                              </Link>
                            )
                          )}

                      </Card.Content>
                    </Card>

                  </Grid.Column>
                </Grid.Row>
              );
            })
          }
      </Grid>
    </React.Fragment>
  );
};

export default Summary;
