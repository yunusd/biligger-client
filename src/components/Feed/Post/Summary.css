.summary-context-right{
  float:right;
}

.card{
  box-shadow:none!important;
}

.display-linebreak {
  white-space: pre-line;
}