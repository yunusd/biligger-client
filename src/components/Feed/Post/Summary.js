import React from 'react';
import { Link } from 'react-router-dom';

import moment from 'moment';
import removeMd from 'remove-markdown';

import { Card, Icon } from 'semantic-ui-react';

import urlSerializer from '../../../helpers/urlSerializer';
import Like from '../../Like';
import dateLocale from '../../../helpers/dateLocale';
import './Summary.css';

moment.updateLocale('en', dateLocale);

const Summary = ({ data, getMe, currentUser }) => data.map((val) => {
  const authorUrl = `/@${val.author.username}`;

  const auth = {
    isOwn: getMe && getMe.username === val.author.username,
    isLoggedIn: currentUser.isLoggedIn && true,
  };

  const title = val.title.length < 100 ? val.title : val.title.slice(0, 100);
  const raw = removeMd(val.content);

  const slug = urlSerializer({
    id: val.id,
    username: val.author.username,
    text: {
      title: val.title,
    },
    type: {
      post: true,
    },
  });


  const content = raw.length < 500 ? raw : `${raw.slice(0, 500)}...`;

  return (
    <Card fluid id={val.id} key={val.id}>
      <Card.Content>
        <Card.Header>
          <Link
            to={{
                  pathname: val.id ? `${slug.post.url}` : '/',
                  state: { id: val.id },
                }}
            style={{ color: 'rgba(0,0,0,.85)', fontSize: '21px' }}
          >
            {title}
          </Link>
        </Card.Header>
        <Card.Meta>
          <Link to={authorUrl}>
            {val.author.username}
          </Link>
                &nbsp;-&nbsp;
          {moment(val.createdAt).fromNow()}
        </Card.Meta>
        <Card.Description style={{ fontSize: '16px', lineHeight: '1.618em' }}>
          {content}
        </Card.Description>
      </Card.Content>

      <Card.Content extra>
        <Like parentModel="Post" id={val.id} like={val.like} />
        &nbsp;&nbsp;&nbsp;
        <Link to="/" className="summary-context-icon">
          <Icon name="comment" size="small" />
        </Link>

        {auth.isLoggedIn && (
          auth.isOwn ? (
            <React.Fragment>
              <Link to={`${slug.post.url}/dÃ¼zenle`} className="summary-context-right summary-context-icon">
                <Icon name="edit" size="small" />
              </Link>
            </React.Fragment>
          ) : (
            <Link to="#" title="Bildir" className="summary-context-right summary-context-icon">
              <Icon name="flag" title="bildir" size="small" />
            </Link>
          )
        )}

      </Card.Content>
    </Card>
  );
});

export default Summary;
