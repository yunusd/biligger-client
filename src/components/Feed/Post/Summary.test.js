import React from 'react';
import { shallow } from 'enzyme';

import Summary from './Summary';

describe('Summary', () => {
  const mock = {
    getLatestPosts: [
      {
        id: '000000000000000000000000',
        title: 'Title',
        content: 'Content',
        author: {
          username: 'Foo',
        },
      },
    ],
  };

  it('render without crashing', () => {
    const wrapper = shallow(<Summary data={mock} />).contains('Foo');
    expect(wrapper).toBeTruthy();
  });

  it('render correctly', () => {
    const wrapper = shallow(<Summary data={mock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
