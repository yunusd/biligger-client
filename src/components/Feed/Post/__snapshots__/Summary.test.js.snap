// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Summary
    data={
      Object {
        "getLatestPosts": Array [
          Object {
            "author": Object {
              "username": "Foo",
            },
            "content": "Content",
            "title": "Title",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "centered": true,
      "children": Array [
        undefined,
        <GridRow>
          <GridColumn
            width={12}
          >
            <Card
              fluid={true}
            >
              <CardContent>
                <CardHeader
                  as="a"
                  href="/"
                >
                  Title
                </CardHeader>
                <CardMeta>
                  Foo
                   - 
                  birkaç saniye önce
                </CardMeta>
                <CardDescription
                  className="display-linebreak"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>Content",
                    }
                  }
                />
              </CardContent>
              <CardContent
                extra={true}
              >
                <Link
                  replace={false}
                  to="/"
                >
                  <Icon
                    as="i"
                    name="idea"
                  />
                  Katılıyorum
                </Link>
                   
                <Link
                  replace={false}
                  to="/"
                >
                  <Icon
                    as="i"
                    name="comment"
                  />
                  Yorum Yaz
                </Link>
                <Link
                  className="summary-context-right"
                  replace={false}
                  to="/"
                >
                  bildir
                </Link>
              </CardContent>
            </Card>
          </GridColumn>
        </GridRow>,
      ],
      "columns": 1,
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridColumn
            width={12}
          >
            <Card
              fluid={true}
            >
              <CardContent>
                <CardHeader
                  as="a"
                  href="/"
                >
                  Title
                </CardHeader>
                <CardMeta>
                  Foo
                   - 
                  birkaç saniye önce
                </CardMeta>
                <CardDescription
                  className="display-linebreak"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>Content",
                    }
                  }
                />
              </CardContent>
              <CardContent
                extra={true}
              >
                <Link
                  replace={false}
                  to="/"
                >
                  <Icon
                    as="i"
                    name="idea"
                  />
                  Katılıyorum
                </Link>
                   
                <Link
                  replace={false}
                  to="/"
                >
                  <Icon
                    as="i"
                    name="comment"
                  />
                  Yorum Yaz
                </Link>
                <Link
                  className="summary-context-right"
                  replace={false}
                  to="/"
                >
                  bildir
                </Link>
              </CardContent>
            </Card>
          </GridColumn>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Card
              fluid={true}
            >
              <CardContent>
                <CardHeader
                  as="a"
                  href="/"
                >
                  Title
                </CardHeader>
                <CardMeta>
                  Foo
                   - 
                  birkaç saniye önce
                </CardMeta>
                <CardDescription
                  className="display-linebreak"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>Content",
                    }
                  }
                />
              </CardContent>
              <CardContent
                extra={true}
              >
                <Link
                  replace={false}
                  to="/"
                >
                  <Icon
                    as="i"
                    name="idea"
                  />
                  Katılıyorum
                </Link>
                   
                <Link
                  replace={false}
                  to="/"
                >
                  <Icon
                    as="i"
                    name="comment"
                  />
                  Yorum Yaz
                </Link>
                <Link
                  className="summary-context-right"
                  replace={false}
                  to="/"
                >
                  bildir
                </Link>
              </CardContent>
            </Card>,
            "width": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <CardContent>
                  <CardHeader
                    as="a"
                    href="/"
                  >
                    Title
                  </CardHeader>
                  <CardMeta>
                    Foo
                     - 
                    birkaç saniye önce
                  </CardMeta>
                  <CardDescription
                    className="display-linebreak"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<p>Content",
                      }
                    }
                  />
                </CardContent>,
                <CardContent
                  extra={true}
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="idea"
                    />
                    Katılıyorum
                  </Link>
                     
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="comment"
                    />
                    Yorum Yaz
                  </Link>
                  <Link
                    className="summary-context-right"
                    replace={false}
                    to="/"
                  >
                    bildir
                  </Link>
                </CardContent>,
              ],
              "fluid": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <CardHeader
                      as="a"
                      href="/"
                    >
                      Title
                    </CardHeader>,
                    <CardMeta>
                      Foo
                       - 
                      birkaç saniye önce
                    </CardMeta>,
                    <CardDescription
                      className="display-linebreak"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<p>Content",
                        }
                      }
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "a",
                      "children": "Title",
                      "href": "/",
                    },
                    "ref": null,
                    "rendered": "Title",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "Foo",
                        " - ",
                        "birkaç saniye önce",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Foo",
                      " - ",
                      "birkaç saniye önce",
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "display-linebreak",
                      "dangerouslySetInnerHTML": Object {
                        "__html": "<p>Content",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Link
                      replace={false}
                      to="/"
                    >
                      <Icon
                        as="i"
                        name="idea"
                      />
                      Katılıyorum
                    </Link>,
                    "   ",
                    <Link
                      replace={false}
                      to="/"
                    >
                      <Icon
                        as="i"
                        name="comment"
                      />
                      Yorum Yaz
                    </Link>,
                    <Link
                      className="summary-context-right"
                      replace={false}
                      to="/"
                    >
                      bildir
                    </Link>,
                  ],
                  "extra": true,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <Icon
                          as="i"
                          name="idea"
                        />,
                        "Katılıyorum",
                      ],
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "i",
                          "name": "idea",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Katılıyorum",
                    ],
                    "type": [Function],
                  },
                  "   ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <Icon
                          as="i"
                          name="comment"
                        />,
                        "Yorum Yaz",
                      ],
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "i",
                          "name": "comment",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Yorum Yaz",
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "bildir",
                      "className": "summary-context-right",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "bildir",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": Array [
          undefined,
          <GridRow>
            <GridColumn
              width={12}
            >
              <Card
                fluid={true}
              >
                <CardContent>
                  <CardHeader
                    as="a"
                    href="/"
                  >
                    Title
                  </CardHeader>
                  <CardMeta>
                    Foo
                     - 
                    birkaç saniye önce
                  </CardMeta>
                  <CardDescription
                    className="display-linebreak"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<p>Content",
                      }
                    }
                  />
                </CardContent>
                <CardContent
                  extra={true}
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="idea"
                    />
                    Katılıyorum
                  </Link>
                     
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="comment"
                    />
                    Yorum Yaz
                  </Link>
                  <Link
                    className="summary-context-right"
                    replace={false}
                    to="/"
                  >
                    bildir
                  </Link>
                </CardContent>
              </Card>
            </GridColumn>
          </GridRow>,
        ],
        "columns": 1,
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridColumn
              width={12}
            >
              <Card
                fluid={true}
              >
                <CardContent>
                  <CardHeader
                    as="a"
                    href="/"
                  >
                    Title
                  </CardHeader>
                  <CardMeta>
                    Foo
                     - 
                    birkaç saniye önce
                  </CardMeta>
                  <CardDescription
                    className="display-linebreak"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<p>Content",
                      }
                    }
                  />
                </CardContent>
                <CardContent
                  extra={true}
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="idea"
                    />
                    Katılıyorum
                  </Link>
                     
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="comment"
                    />
                    Yorum Yaz
                  </Link>
                  <Link
                    className="summary-context-right"
                    replace={false}
                    to="/"
                  >
                    bildir
                  </Link>
                </CardContent>
              </Card>
            </GridColumn>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Card
                fluid={true}
              >
                <CardContent>
                  <CardHeader
                    as="a"
                    href="/"
                  >
                    Title
                  </CardHeader>
                  <CardMeta>
                    Foo
                     - 
                    birkaç saniye önce
                  </CardMeta>
                  <CardDescription
                    className="display-linebreak"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<p>Content",
                      }
                    }
                  />
                </CardContent>
                <CardContent
                  extra={true}
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="idea"
                    />
                    Katılıyorum
                  </Link>
                     
                  <Link
                    replace={false}
                    to="/"
                  >
                    <Icon
                      as="i"
                      name="comment"
                    />
                    Yorum Yaz
                  </Link>
                  <Link
                    className="summary-context-right"
                    replace={false}
                    to="/"
                  >
                    bildir
                  </Link>
                </CardContent>
              </Card>,
              "width": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <CardContent>
                    <CardHeader
                      as="a"
                      href="/"
                    >
                      Title
                    </CardHeader>
                    <CardMeta>
                      Foo
                       - 
                      birkaç saniye önce
                    </CardMeta>
                    <CardDescription
                      className="display-linebreak"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<p>Content",
                        }
                      }
                    />
                  </CardContent>,
                  <CardContent
                    extra={true}
                  >
                    <Link
                      replace={false}
                      to="/"
                    >
                      <Icon
                        as="i"
                        name="idea"
                      />
                      Katılıyorum
                    </Link>
                       
                    <Link
                      replace={false}
                      to="/"
                    >
                      <Icon
                        as="i"
                        name="comment"
                      />
                      Yorum Yaz
                    </Link>
                    <Link
                      className="summary-context-right"
                      replace={false}
                      to="/"
                    >
                      bildir
                    </Link>
                  </CardContent>,
                ],
                "fluid": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <CardHeader
                        as="a"
                        href="/"
                      >
                        Title
                      </CardHeader>,
                      <CardMeta>
                        Foo
                         - 
                        birkaç saniye önce
                      </CardMeta>,
                      <CardDescription
                        className="display-linebreak"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<p>Content",
                          }
                        }
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": "a",
                        "children": "Title",
                        "href": "/",
                      },
                      "ref": null,
                      "rendered": "Title",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          "Foo",
                          " - ",
                          "birkaç saniye önce",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Foo",
                        " - ",
                        "birkaç saniye önce",
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "className": "display-linebreak",
                        "dangerouslySetInnerHTML": Object {
                          "__html": "<p>Content",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Link
                        replace={false}
                        to="/"
                      >
                        <Icon
                          as="i"
                          name="idea"
                        />
                        Katılıyorum
                      </Link>,
                      "   ",
                      <Link
                        replace={false}
                        to="/"
                      >
                        <Icon
                          as="i"
                          name="comment"
                        />
                        Yorum Yaz
                      </Link>,
                      <Link
                        className="summary-context-right"
                        replace={false}
                        to="/"
                      >
                        bildir
                      </Link>,
                    ],
                    "extra": true,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": Array [
                          <Icon
                            as="i"
                            name="idea"
                          />,
                          "Katılıyorum",
                        ],
                        "replace": false,
                        "to": "/",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "as": "i",
                            "name": "idea",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "Katılıyorum",
                      ],
                      "type": [Function],
                    },
                    "   ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": Array [
                          <Icon
                            as="i"
                            name="comment"
                          />,
                          "Yorum Yaz",
                        ],
                        "replace": false,
                        "to": "/",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "as": "i",
                            "name": "comment",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "Yorum Yaz",
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "bildir",
                        "className": "summary-context-right",
                        "replace": false,
                        "to": "/",
                      },
                      "ref": null,
                      "rendered": "bildir",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
